include/wx/dc.h

```diff
@@ -1542,5 +1542,41 @@ class WXDLLIMPEXP_CORE wxDCFontChanger
     wxDECLARE_NO_COPY_CLASS(wxDCFontChanger);
 };
 
+// ----------------------------------------------------------------------------
+// helper class: you can use it to save|temporarily change the DC background mode,
+// and restore it automatically, when the object goes out of scope
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxDCBgModeChanger
+{
+public:
+  wxDCBgModeChanger(wxDC& dc)
+    : m_dc(dc), m_bgmOld(dc.GetBackgroundMode())
```

Comment:
`m_bgmOld` should be initialized to `wxBRUSHSTYLE_INVALID` in this ctor and the dtor should check if it's not set to it to avoid restoring the background mode unnecessarily if it had never been changed.